internal/docker/docker_checks.go

```diff
@@ -21,27 +21,33 @@ import (
 	"os/exec"
 )
 
-// CheckDockerConfig is a function to check docker and docker-compose configuration on the host
-func CheckDockerConfig() error {
+func CheckDockerConfig() (DockerComposeVersion, error) {
 
 	dockerCmd := exec.Command("docker", "-v")
 	_, err := dockerCmd.Output()
 	if err != nil {
-		return fmt.Errorf("an error occurred while running docker. Is docker installed on your computer?")
-	}
-
-	dockerComposeCmd := exec.Command("docker-compose", "-v")
-	_, err = dockerComposeCmd.Output()
-
-	if err != nil {
-		return fmt.Errorf("an error occurred while running docker-compose. Is docker-compose installed on your computer?")
+		return None, fmt.Errorf("an error occurred while running docker. Is docker installed on your computer?")
 	}
 
 	dockerDeamonCheck := exec.Command("docker", "ps")
 	_, err = dockerDeamonCheck.Output()
 	if err != nil {
-		return fmt.Errorf("an error occurred while running docker. Is docker running on your computer?")
+		return None, fmt.Errorf("an error occurred while running docker. Is docker running on your computer?")
 	}
 
-	return nil
+	// ckeck for docker-compose (v1) version
+	dockerComposeV1Cmd := exec.Command("docker-compose", "-v")
+	_, errV1 := dockerComposeV1Cmd.Output()
+
+	// ckeck for docker-compose (V2) version
+	dockerComposeV2Cmd := exec.Command("docker compose", "version")
+	_, errV2 := dockerComposeV2Cmd.Output()
+	
+	if errV1 == nil && errV2 != nil {
+		return ComposeV1, nil
+	} else if errV1 != nil && errV2 == nil {
+		return ComposeV2, nil
+	} else {
+		return None, fmt.Errorf("an error occurred while running docker-compose. Is docker-compose installed on your computer?")
+	}
```

Comment:
I also might suggest a `switch`/`case` statement here instead of several `else if`s. In Go, you don't actually have to `switch` "on" a variable.

Reply:
Yes, having both installed is one of the tests missing. Although, handling that case is quite intriguing. My intuition would be to force the usage of the new one. Not out of disregard for stability but to ensure that users, without any settings or configuration, are set to use the newer future proof api docker provides. This in turn also leads to bugs being found that pertain to the switch between v1 and v2. Less radical is keeping v1 as the dominant, when both versions are present.

Reply:
I think another solution would be to keep the flow of the application and instead of using my first draft of convoluted checks to a more elegant solution;

```go
	// ckeck for docker-compose (v1) version
	dockerComposeCmd := exec.Command("docker-compose", "-v")
	_, err = dockerComposeCmd.Output()
	if err == nil {
		return ComposeV1, nil
	}

	// ckeck for docker-compose (V2) version
	dockerComposeCmd = exec.Command("docker compose", "version")
	_, err = dockerComposeCmd.Output()
	if err == nil {
		return ComposeV2, nil
	}

	return None, fmt.Errorf("an error occurred while running docker-compose. Is docker-compose installed on your computer?")
```

The checks are in sequence with the first check being the preferred if both were to be present.

Reply:
Another more complex solution is to allow the user to specify which docker-version he would like to use to invoke firefly-cli. But this would require a flag or parameter to be included in the available api of firefly-cli. I see this as a more extensive change and not in scope for this PR.